Funcky/Extensions/EnumerableExtensions/PartitionBuilder`2.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Immutable;
+
+namespace Funcky.Extensions;
+
+internal readonly struct PartitionBuilder<TLeft, TRight>
```

Comment:
What about using a readonly record struct, you write your own `With` and you could avoid the default values in the constructor.